katharsis-jpa/src/main/java/io/katharsis/jpa/internal/meta/impl/MetaKeyImpl.java

```diff
@@ -0,0 +1,136 @@
+package io.katharsis.jpa.internal.meta.impl;
+
+import java.util.List;
+
+import io.katharsis.jpa.internal.meta.MetaAttribute;
+import io.katharsis.jpa.internal.meta.MetaEmbeddable;
+import io.katharsis.jpa.internal.meta.MetaKey;
+import io.katharsis.jpa.internal.meta.MetaType;
+
+public class MetaKeyImpl extends MetaElementImpl implements MetaKey {
+
+	private static final String ID_ELEMENT_SEPARATOR = "-";
+
+	private String name;
+	private List<MetaAttribute> elements;
+	private boolean primaryKey;
+	private boolean unique;
+	private MetaType type;
+
+	public MetaKeyImpl(MetaDataObjectImpl parent, String name, List<MetaAttribute> elements, boolean primaryKey,
+			boolean unique, MetaType type) {
+		super(parent);
+		this.name = name;
+		this.elements = elements;
+		this.primaryKey = primaryKey;
+		this.unique = unique;
+		this.type = type;
+	}
+
+	@Override
+	public String getId() {
+		return getParent().getId() + "." + getName();
+	}
+
+	@Override
+	public List<MetaAttribute> getElements() {
+		return elements;
+	}
+
+	@Override
+	public boolean isUnique() {
+		return unique;
+	}
+
+	@Override
+	public boolean isPrimaryKey() {
+		return primaryKey;
+	}
+
+	@Override
+	public String getName() {
+		return name;
+	}
+
+	@Override
+	public MetaType getType() {
+		return type;
+	}
+
+	@Override
+	public MetaAttribute getUniqueElement() {
+		if (elements.size() != 1)
+			throw new IllegalStateException(getName() + " must contain a single primary key attribute");
+		return elements.get(0);
+	}
+
+	@Override
+	public Object fromKeyString(String idString) {
+
+		// TODO could be more sophisticated
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)